pkg/controllers/state/cluster.go

```diff
@@ -419,6 +422,15 @@ func (c *Cluster) MarkPodSchedulingDecisions(ctx context.Context, podErrors map[
 			}
 		}
 	}
+	c.UpdatePodToNodeClaimMapping(ncPods)
+}
+
+func (c *Cluster) UpdatePodToNodeClaimMapping(ncPods map[string][]*corev1.Pod) {
```

Comment:
nit: You could possibly consider inlining this in MarkPodSchedulingDecisions and just add a comment to it

Reply:
Causes problem with the cyclomatic complexity of the function, hence did not implement it.